server/graphql/utils/request.ts

```diff
@@ -20,19 +20,68 @@ const getRequestsById = (requestIds, dataSources) => {
   return requestIds.map(id => dataSources.requests.getById(id))
 }
 
-const updateRequestHelper = (request, dataSources): Promise<ServerResponseInterface> => {
+const updateRequestHelper = (request, dataSources): Promise<Document> => {
   if(!request.id) {
     throw new UserInputError('Missing argument value', { argumentName: 'id' })
   }
-  const requestTypeId = dataSources.requests.getById(request.id.toString()).requestType
-  return dataSources.requests.update(request)
-    .then(res => {
-      updateRequestTypeHelper({"id": requestTypeId}, dataSources)
-      return res
-    })
-    .catch(error => {
-      return error
-    })
+
+  const currentRequestCopy = dataSources.requests.getById(request.id.toString()).toObject()
+  currentRequestCopy.id = currentRequestCopy._id
+  const currentRequest = dataSources.requests.getById(request.id.toString())
+  let oldRequestTypeCopy, newRequestTypeCopy
+
+  if(request.requestType) {
+    // If these fail then nothing has been changed at this point, so nothing to revert
+    oldRequestTypeCopy = dataSources.requestTypes.getById(currentRequest.requestType.toString()).toObject()
+    oldRequestTypeCopy.id = oldRequestTypeCopy._id
+    newRequestTypeCopy = dataSources.requestTypes.getById(request.requestType.toString()).toObject()
+    newRequestTypeCopy.id = newRequestTypeCopy._id
+  }
+
+  if(request.fulfilled === true) {
```

Comment:
We should set `dateFulfilled` when `request.fulfilled === true && currentRequest.fulfilled === false`

Reply:
Updated.